src/psyclone/tests/psyir/frontend/fparser2_not_init_test.py

```diff
@@ -0,0 +1,73 @@
+# -----------------------------------------------------------------------------
+# BSD 3-Clause License
+#
+# Copyright (c) 2021, Science and Technology Facilities Council.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author: J. Henrichs, Bureau of Meteorology
+
+''' Module containing pytest test that verifies proper behaviour if
+fparser was not initialised.'''
+
+from __future__ import absolute_import
+
+import pytest
+
+from fparser.two.parser import ParserFactory
+
+from psyclone.gocean1p0 import GOLoop
+
+
+@pytest.fixture(scope="function", autouse=True)
+def clear_fparser():
+    ''' The tests here assume that fparser has not been initialised.
+    This is achieved by calling `_setup([])` with an empty list, which
+    will remove all currently existing parser classes and functions.
+    At the end of the tests re-initialse parser.
+    '''
+
+    # Remove all fparser classes and functions
+    ParserFactory()._setup([])
+
+    # Now execute all tests
+    yield
+
+    # We need to properly initialise fparser,
+    # otherwise followup tests will fail (if this test should fail)
+    ParserFactory().create(std="f2008")
+
+
+def test_loop_bound_when_fparser_not_initialised():
```

Comment:
Thanks for adding a new file. However, since this is a test of GOLoop functionality, I think it should live in a GOcean-appropriate location (maybe `tests/domain/gocean/goloop_test.py`?) rather than in the PSyIR frontend. You'll need to move the fixture too and turn off its 'autouse'. As you say, the fixture could be useful in general so please could you put it in `conftest.py` and add some words about it to the section on fixtures in the dev guide.

Reply:
Done, but I put the fixture in goloop_test. I think for a fixture which atm is only used once, it should be very close to the test where it is used (adding something to conftest imho indicates it is used more than once). This way the whole program flow is easily visible and well documented on one screen.

When we get additional tests that need this fixture, it should be moved.

Reply:
OK, thanks.